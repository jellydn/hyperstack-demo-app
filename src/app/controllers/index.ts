// naming and imports: name and import as you see fit
import Auth from './auth'
import { NotesController } from './notes'

const controllers = []
controllers.push(Auth)
controllers.push(NotesController)

export const AppController = {
  controllers,
}
