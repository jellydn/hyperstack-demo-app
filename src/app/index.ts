import type { Context } from 'hyperstack'
import { context } from 'hyperstack'
import type * as Models from './models'
import type * as Mailers from './mailers'

export const appContext = context as Context<typeof Models, typeof Mailers>
