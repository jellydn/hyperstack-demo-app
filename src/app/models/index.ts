export { Note } from './note'
export { User } from './user'
