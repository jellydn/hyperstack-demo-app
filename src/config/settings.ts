// general purpose settings, with no infrastructure loading, no assumptions. keep it simple.
import path from 'path'
export const root = path.join(__dirname, '..')
