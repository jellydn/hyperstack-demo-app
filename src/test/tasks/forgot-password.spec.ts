import { test } from '@hyperstackjs/testing'
import ForgotPassword from '../../lib/tasks/forgot-password'
import { appContext } from '../../app'

import { root } from '../../config/settings'

const { tasks } = test(root)

describe('tasks', () => {
  describe('forgot-password', () => {
    tasks('should work', async (app) => {
      const { User } = appContext.models()
      const user = await User.createWithPassword({
        username: 'evh@example.com',
        password: 'mypass-should-login',
        name: 'Eddie Van Halen',
      })
      expect(
        await ForgotPassword.exec({ username: user.username }, app)
      ).toMatchSnapshot()
    })
  })
})
