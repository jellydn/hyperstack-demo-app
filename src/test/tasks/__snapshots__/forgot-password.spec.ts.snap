// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tasks forgot-password should work 1`] = `
{
  "ok": true,
}
`;
